Tasks/FileTransformV2/task.json

```diff
@@ -0,0 +1,121 @@
+{
+    "id": "8ce97e91-56cc-4743-bfab-9a9315be5f27",
+    "name": "FileTransform",
+    "friendlyName": "File transform",
+    "description": "Replace tokens with variable values in XML or JSON configuration files",
+    "author": "Microsoft Corporation",
+    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/file-transform",
+    "helpMarkDown": "File transformation and variable substitution task: Update tokens in your XML based configuration files and then replaces those tokens with variable values. <br/>Currently only XML, JSON file formats are supported for variable substitution.",
+    "category": "Utility",
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "runsOn": [
+        "Agent",
+        "DeploymentGroup"
+    ],
+    "version": {
+        "Major": 2,
+        "Minor": 0,
+        "Patch": 0
+    },
+    "instanceNameFormat": "File Transform: $(Package)",
+    "groups": [
+        {
+            "name": "VariableSubstitution",
+            "displayName": "Variable Substitution",
+            "isExpanded": true
+        }
+    ],
+    "inputs": [
+        {
+            "name": "folderPath",
+            "type": "filePath",
+            "label": "Package or folder",
+            "defaultValue": "$(System.DefaultWorkingDirectory)/**/*.zip",
+            "required": true,
+            "helpMarkDown": "File path to the package or a folder.<br />Variables ( [Build](https://docs.microsoft.com/vsts/pipelines/build/variables) | [Release](https://docs.microsoft.com/vsts/pipelines/release/variables#default-variables)), wildcards are supported. <br/> For example, $(System.DefaultWorkingDirectory)/\\*\\*/\\*.zip."
+        },
+        {
+            "name": "enableXmlTransform",
+            "type": "boolean",
+            "label": "XML transformation",
+            "required": false,
+            "defaultValue": false,
+            "helpMarkDown": "Config transforms will be run prior to the Variable Substitution.<br/>XML transformations are supported only for Windows platform."
+        },
+        {
+            "name": "xmlTransformationRules",
+            "type": "multiLine",
+            "label": "Transformation rules",
+            "defaultValue": "-transform **\\*.Release.config -xml **\\*.config",
+            "required": true,
+            "helpMarkDown": "Provide new line separated list of transformation file rules using the syntax: <br/>-transform <pathToTransformFile>  -xml <pathToSourceConfigurationFile>",
+            "visibleRule": "enableXmlTransform == true"
+        },
+        {
+            "name": "fileType",
```

Comment:
@N-Usha , Can you please reconsider spliiting XML and JSON into separate inputs.
This way, if a user can use single task to transform XML and JSON .

Reply:
Sure, we could consider that. But the intent was to keep this task generic enough to handle even other file formats (YAML in particular) for variable substitution. We should then think of a multi-select component or check-boxes for each option. Coz once you select a file format, you would also need another field input for target files per format.. Let me explore and revert about how the designer and YAML syntax could look in that option.